extendeddatacrossword_test.go

```diff
@@ -423,6 +424,186 @@ func TestCorruptedEdsReturnsErrByzantineData_UnorderedShares(t *testing.T) {
 	}
 }
 
+func TestFuzzRandByzantine(t *testing.T) {
+	// This test is slow and should be skipped during normal testing
+	t.Skip()
+	for i := 0; i < 10000; i++ {
+		TestErrRandByzantine(t)
+	}
+}
+
+func TestErrRandByzantine(t *testing.T) {
+	codec := NewLeoRSCodec()
+	original, corrupted, idx := randCorruptedEDS(t, codec, 8)
+	require.False(t, original.Equals(corrupted), "corrupted eds is equal to original eds")
+
+	newEds, err := repairNewFromCorrupted(codec, corrupted, idx)
+	if err != nil && newEds != nil {
+		// visual check of the new eds
+		prettyPrintEds(newEds)
+		fmt.Println("new eds is original", original.Equals(newEds))
+		fmt.Println("new eds is corrupted", corrupted.Equals(newEds))
+	}
+	require.NoError(t, err, "failure to reconstruct the extended data square")
+}
+
+func randCorruptedEDS(t require.TestingT, codec Codec, size int) (original, corrupted *ExtendedDataSquare, idx int) {
+	ds := genRandDS(size, shareSize)
+	original, err := ComputeExtendedDataSquare(ds, codec, NewDefaultTree)
+	require.NoError(t, err)
+
+	// create random share
+	randShare := make([]byte, shareSize)
+	_, _ = crand.Read(randShare)
+
+	// choose a random share to corrupt
+	shares := original.Flattened()
+	idx = rand.Intn(len(shares))
+
+	// copy namespace to avoid namespace ordering issues
+	copy(randShare, shares[idx][:nmt.DefaultNamespaceIDLen])
+
+	// corrupt the share
+	shares[idx] = randShare
+
+	corrupted, err = ImportExtendedDataSquare(
+		shares,
+		codec,
+		NewDefaultTree)
+	require.NoError(t, err)
+	return original, corrupted, idx
+}
+
+func repairNewFromCorrupted(codec Codec, corrupted *ExtendedDataSquare, corruptedIdx int) (*ExtendedDataSquare, error) {
+	samples := make([][]bool, corrupted.Width())
+	for i := range samples {
+		samples[i] = make([]bool, corrupted.Width())
+	}
+
+	square, err := NewExtendedDataSquare(
+		codec,
+		NewDefaultTree,
+		corrupted.Width(),
+		shareSize,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failure to create extended data square: %w", err)
+	}
+
+	// set corrupted share first
+	errX, errY := corruptedIdx/int(corrupted.Width()), corruptedIdx%int(corrupted.Width())
+	share := corrupted.GetCell(uint(errX), uint(errY))
+	err = square.SetCell(uint(errX), uint(errY), share)
+	if err != nil {
+		return nil, fmt.Errorf("failure to set corrupted share: %w", err)
+	}
+
+	rowRoots, err := corrupted.RowRoots()
+	if err != nil {
+		return nil, fmt.Errorf("failure to get row roots: %w", err)
+	}
+	colRoots, err := corrupted.ColRoots()
+	if err != nil {
+		return nil, fmt.Errorf("failure to get column roots: %w", err)
+	}
+
+	// loop until repaired or byzantine error
+	for {
+		repaired, err := fillRandomCellAndRepair(corrupted, square, rowRoots, colRoots, samples)
+		if repaired {
+			prettyPrintSamples(samples, corruptedIdx)
+			return square, errors.New("no byzantine error")
```

Comment:
[question] why return an error her? Why not

```suggestion
			return square, nil
```

Reply:
Test corrupts eds and tries to repair it. It shoud result in Byzantine every time on corrupted eds. If there was no encoding errors during eds construction, it means we missed some cases.